### ISSUE TITLE
Content Assist sometimes not Picking up local workfields

### ISSUE THREAD

**Issue by Wikus86** [codefori/vscode-rpgle#294]
Here is an exmaple of it not picking up the local workfield.

![image](https://github.com/codefori/vscode-rpgle/assets/36194434/fbd62d2c-8f10-4268-8aa1-39ce4e0a1d54)

Another thing i noticed, linter picks up that it has not been references

![image](https://github.com/codefori/vscode-rpgle/assets/36194434/f8f85158-083c-423f-b244-24325d6578c1)

**Comment by worksofliam** [comment]
@Wikus86 I have tried recreating this with no luck.

Are you able to paste your code (not take a screenshot of it) so I can copy and paste it directly?

<img width="446" alt="image" src="https://github.com/codefori/vscode-rpgle/assets/3708366/e9de882f-6c69-4c86-85d5-404721f10aba">

Thanks,
Liam

**Comment by Wikus86** [comment]
Yeah, I can also not reproduce it, only happens in some programs. Let me try doing a debug on the program giving the issue. Will advise if I find something

**Comment by Wikus86** [comment]
I found something, looks like it gets confused with a procedure in my copybook.

This is an example of the code

![image](https://github.com/codefori/vscode-rpgle/assets/36194434/d2f054dc-f75a-4576-a95e-95ab689c9597)

in **completionItem.ts**, in the **completionItemProvider** function, it picks up i am on line 318 which is almost correct, actually on line 319

![Screenshot 2024-01-30 112042](https://github.com/codefori/vscode-rpgle/assets/36194434/8d6b3720-21df-43e8-ab13-1b63dfcbba38)

Then it tries to find the procedure based on the line number

![image](https://github.com/codefori/vscode-rpgle/assets/36194434/a46a7f9a-0e6c-4538-b678-91e42fe1d8ad)


The problem is, in my copy book I have a procedure in that line range

![Screenshot 2024-01-30 111944](https://github.com/codefori/vscode-rpgle/assets/36194434/1345e97a-cfc3-4a33-87f8-ec723059fea7)


and the index is before the actual procedure I am in

![Screenshot 2024-01-30 112015](https://github.com/codefori/vscode-rpgle/assets/36194434/622fb974-6c7a-4de4-8521-9959b5573ba4)

I hope this makes a bit of sense :)

**Comment by EVLSDE** [comment]
> The problem is, in my copy book I have a procedure in that line range

I just stumbled over this sentence. Could it be I accidentally posted a duplicate issue in #323? If yes then I'm sorry!
The code in pull request #324 worked it out for me.

**Comment by worksofliam** [comment]
Looks like this issue might have been solved by #324.
